src/types/ColorFix.cpp

```diff
@@ -1,231 +1,186 @@
+            -1.2684380046f * l + 2.6097574011f * m - 0.3413193965f * s,
+            -0.0041960863f * l - 0.7034186147f * m + 1.7076147010f * s,
+        };
+    }
+}
 
-    // Put it all together!
-    const auto lightness = deltaLPrime / (kSubL * sSubL);
-    const auto chroma = deltaCPrime / (kSubC * sSubC);
-    const auto hue = deltaHPrime / (kSubH * sSubH);
+#pragma warning(push)
+#pragma warning(disable : 26446) // Prefer to use gsl::at() instead of unchecked subscript operator (bounds.4).
+#pragma warning(disable : 26482) // Only index into arrays using constant expressions (bounds.2).
 
-    return sqrtf(powf(lightness, 2) + powf(chroma, 2) + powf(hue, 2) + rSubT * chroma * hue);
+static oklab::RGB colorrefToLinear(COLORREF c) noexcept
+{
+    const auto r = srgbToRgbLUT[(c >> 0) & 0xff];
+    const auto g = srgbToRgbLUT[(c >> 8) & 0xff];
+    const auto b = srgbToRgbLUT[(c >> 16) & 0xff];
+    return { r, g, b };
 }
 
-// Method Description:
-// - Populates our L, A, B values, based on our r, g, b values
-// - Converts a color in rgb format to a color in lab format
-// - Reference: http://www.easyrgb.com/index.php?X=MATH&H=01#text1
-void ColorFix::_ToLab() noexcept
+#pragma warning(pop)
+
+static COLORREF linearToColorref(const oklab::RGB& c) noexcept
 {
-    auto var_R = r / 255.0f;
-    auto var_G = g / 255.0f;
-    auto var_B = b / 255.0f;
-
-    var_R = var_R > 0.04045f ? powf(((var_R + 0.055f) / 1.055f), 2.4f) : var_R / 12.92f;
-    var_G = var_G > 0.04045f ? powf(((var_G + 0.055f) / 1.055f), 2.4f) : var_G / 12.92f;
-    var_B = var_B > 0.04045f ? powf(((var_B + 0.055f) / 1.055f), 2.4f) : var_B / 12.92f;
-
-    var_R = var_R * 100.0f;
-    var_G = var_G * 100.0f;
-    var_B = var_B * 100.0f;
-
-    //Observer. = 2 degrees, Illuminant = D65
-    const auto X = var_R * 0.4124f + var_G * 0.3576f + var_B * 0.1805f;
-    const auto Y = var_R * 0.2126f + var_G * 0.7152f + var_B * 0.0722f;
-    const auto Z = var_R * 0.0193f + var_G * 0.1192f + var_B * 0.9505f;
-
-    auto var_X = X / 95.047f; //ref_X =  95.047   (Observer= 2 degrees, Illuminant= D65)
-    auto var_Y = Y / 100.000f; //ref_Y = 100.000
-    auto var_Z = Z / 108.883f; //ref_Z = 108.883
-
-    var_X = var_X > 0.008856f ? powf(var_X, (1.0f / 3.0f)) : (7.787f * var_X) + (16.0f / 116.0f);
-    var_Y = var_Y > 0.008856f ? powf(var_Y, (1.0f / 3.0f)) : (7.787f * var_Y) + (16.0f / 116.0f);
-    var_Z = var_Z > 0.008856f ? powf(var_Z, (1.0f / 3.0f)) : (7.787f * var_Z) + (16.0f / 116.0f);
-
-    L = (116.0f * var_Y) - 16.0f;
-    A = 500.0f * (var_X - var_Y);
-    B = 200.0f * (var_Y - var_Z);
+    auto r = saturate(c.r);
+    auto g = saturate(c.g);
+    auto b = saturate(c.b);
+    r = (r > 0.0031308f ? (1.055f * powf(r, 1.0f / 2.4f) - 0.055f) : 12.92f * r) * 255.0f;
+    g = (g > 0.0031308f ? (1.055f * powf(g, 1.0f / 2.4f) - 0.055f) : 12.92f * g) * 255.0f;
+    b = (b > 0.0031308f ? (1.055f * powf(b, 1.0f / 2.4f) - 0.055f) : 12.92f * b) * 255.0f;

```

Comment:
Fabian Giesen (ryg) has also written some great linear -> sRGB routines and published them here: https://gist.github.com/rygorous/2203834
But I felt like that's a bit too soon for now.